scripts.js

```diff
@@ -58,15 +58,8 @@ function randomizeColorPalette() {
 
 function lockColor(event) {
   var boxId = event.target.id;
-  for (var i = 0; i < currentPalette.colors.length; i++) {
-    console.log(boxId, currentPalette.colors[i].id);
-    if (boxId === currentPalette.colors[i].id) {
-      currentPalette.colors[i].isLocked = !currentPalette.colors[i].isLocked;
-      locks[i].innerText = currentPalette.colors[i].isLocked
-        ? "lock"
-        : "lock_open";
-    }
-  }
+  var index = currentPalette.toggleColorLock(boxId)
+  locks[index].innerText = currentPalette.colors[index].lockIconValue()
```

Comment:
Move lines `61-64` to be in a method called `toggleColorLock` in the palette class. This class will then do the logic of lines `61-64` and will return `i` from the for loop. This will be our index for locks in order for us to target the correct lock for line `62`. `.lockIconValue` took lines `65-68` and made a method inside the color class since our condition is based on the property `isLocked` in color. It's more OOP to have them live together in the same class.